crates/kitsune2/src/factories/mem_peer_store.rs

```diff
@@ -0,0 +1,261 @@
+//! A production-ready memory-based peer store.
+
+use kitsune2_api::{agent::*, config::*, peer_store::*, *};
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+/// Configuration parameters for [MemPeerStoreFactory]
+#[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MemPeerStoreConfig {}
+
+impl ModConfig for MemPeerStoreConfig {}
+
+/// A production-ready memory-based peer store factory.
+#[derive(Debug)]
+pub struct MemPeerStoreFactory {}
+
+impl MemPeerStoreFactory {
+    /// Construct a new MemPeerStoreFactory
+    pub fn create() -> DynPeerStoreFactory {
+        let out: DynPeerStoreFactory = Arc::new(Self {});
+        out
+    }
+}
+
+impl PeerStoreFactory for MemPeerStoreFactory {
+    fn default_config(&self, _config: &mut Config) {}
+
+    fn create(
+        &self,
+        _builder: Arc<builder::Builder>,
+    ) -> BoxFut<'static, K2Result<DynPeerStore>> {
+        Box::pin(async move {
+            let out: DynPeerStore = Arc::new(MemPeerStore::new());
+            Ok(out)
+        })
+    }
+}
+
+struct MemPeerStore(Mutex<Inner>);
+
+impl std::fmt::Debug for MemPeerStore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MemPeerStore").finish()
+    }
+}
+
+impl MemPeerStore {
+    pub fn new() -> Self {
+        Self(Mutex::new(Inner::new(std::time::Instant::now())))
+    }
+}
+
+impl peer_store::PeerStore for MemPeerStore {
+    fn insert(
+        &self,
+        agent_list: Vec<Arc<AgentInfoSigned>>,
+    ) -> BoxFut<'_, K2Result<()>> {
+        self.0.lock().unwrap().insert(agent_list);
+        Box::pin(async move { Ok(()) })
+    }
+
+    fn get(
+        &self,
+        agent: AgentId,
+    ) -> BoxFut<'_, K2Result<Option<Arc<AgentInfoSigned>>>> {
+        let r = self.0.lock().unwrap().get(agent);
+        Box::pin(async move { Ok(r) })
+    }
+
+    fn get_all(&self) -> BoxFut<'_, K2Result<Vec<Arc<AgentInfoSigned>>>> {
+        let r = self.0.lock().unwrap().get_all();
+        Box::pin(async move { Ok(r) })
+    }
+
+    fn get_overlapping_storage_arc(
+        &self,
+        arc: BasicArc,
+    ) -> BoxFut<'_, K2Result<Vec<Arc<AgentInfoSigned>>>> {
+        let r = self.0.lock().unwrap().get_overlapping_storage_arc(arc);
+        Box::pin(async move { Ok(r) })
+    }
+
+    fn get_near_location(
+        &self,
+        loc: u32,
+        limit: usize,
+    ) -> BoxFut<'_, K2Result<Vec<Arc<AgentInfoSigned>>>> {
+        let r = self.0.lock().unwrap().get_near_location(loc, limit);
+        Box::pin(async move { Ok(r) })
+    }
+}
+
+struct Inner {
+    store: HashMap<AgentId, Arc<AgentInfoSigned>>,
+    no_prune_until: std::time::Instant,
+}
+
+impl Inner {
+    pub fn new(now_inst: std::time::Instant) -> Self {
+        Self {
+            store: HashMap::new(),
+            no_prune_until: now_inst + std::time::Duration::from_secs(10),
```

Comment:
Should this be configurable?

Reply:
Done: https://github.com/holochain/kitsune2/pull/19/commits/cc03e5be6c53848dec1c5c8e3fc5d500cb774e84